src/test/components/Viewport.test.js

```diff
@@ -0,0 +1,601 @@
+      });
+    });
+
+    it('will anchor small content at the top', function() {
+      const { getChartProps, clickAndDrag } = setup({
+        maxViewportHeight: SMALL_MAX_VIEWPORT_HEIGHT,
+        startsAtBottom: false,
+      });
+      const anchoredViewport = {
+        viewportLeft: 0,
+        viewportRight: 1,
+        viewportTop: 0,
+        viewportBottom: COMPONENT_HEIGHT,
+      };
+      expect(getChartProps().viewport).toMatchObject(anchoredViewport);
+
+      // It will not move the viewport.
+      clickAndDrag(middleX, middleY, middleX, middleY - 10);
+      expect(getChartProps().viewport).toMatchObject(anchoredViewport);
+      clickAndDrag(middleX, middleY, middleX, middleY + 10);
+      expect(getChartProps().viewport).toMatchObject(anchoredViewport);
+    });
+
+    it('will anchor small content at the bottom', function() {
+      const { getChartProps, clickAndDrag } = setup({
+        maxViewportHeight: SMALL_MAX_VIEWPORT_HEIGHT,
+        startsAtBottom: true,
+      });
+      const anchoredViewport = {
+        viewportLeft: 0,
+        viewportRight: 1,
+        viewportTop: SMALL_MAX_VIEWPORT_HEIGHT - COMPONENT_HEIGHT,
+        viewportBottom: SMALL_MAX_VIEWPORT_HEIGHT,
+      };
+      expect(getChartProps().viewport).toMatchObject(anchoredViewport);
+
+      // It will not move the viewport.
+      clickAndDrag(middleX, middleY, middleX, middleY - 10);
+      expect(getChartProps().viewport).toMatchObject(anchoredViewport);
+      clickAndDrag(middleX, middleY, middleX, middleY + 10);
+      expect(getChartProps().viewport).toMatchObject(anchoredViewport);
+    });
+
+    it('can click and drag left/right', function() {
+      const { scrollAndGetViewport, getChartProps, clickAndDrag } = setup();
+
+      // Assert the initial values.
+      const { viewportLeft, viewportRight } = scrollAndGetViewport({
+        // Zoom in some large arbitrary amount:
+        deltaY: -5000,
+        shiftKey: true,
+        clientX: LEFT + COMPONENT_WIDTH * 0.5,
+      });
+
+      // These values are arbitrary, but show that the viewport was zoomed in.
+      expect(viewportLeft).toBeGreaterThan(0.3);
+      expect(viewportRight).toBeLessThan(0.7);
+
+      // Perform the dragging action.
+      clickAndDrag(middleX, middleY, middleX + 500, middleY);
+      expect(getChartProps().viewport.viewportLeft).toBeLessThan(viewportLeft);
+      expect(getChartProps().viewport.viewportRight).toBeLessThan(
+        viewportRight
+      );
+
+      // Drag back the other way.
+      clickAndDrag(middleX, middleY, middleX - 1000, middleY);
+      expect(getChartProps().viewport.viewportLeft).toBeGreaterThan(
+        viewportLeft
+      );
+      expect(getChartProps().viewport.viewportRight).toBeGreaterThan(
+        viewportRight
+      );
+    });
+
+    it('will not scroll off to the left of the viewport bounds', function() {
+      const { scrollAndGetViewport, getChartProps, clickAndDrag } = setup();
+      scrollAndGetViewport({
```

Comment:
nit: you don't actually use the returned viewport, so just use the `scroll` utility.
(and the same below)